activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -104,6 +104,33 @@ def take!
       take || raise_record_not_found_exception!
     end
 
+    # Finds the sole matching record. Raises ActiveRecord::RecordNotFound if no
+    # record is found. Raises ActiveRecord::SoleRecordExceeded if more than one
+    # record is found.
+    #
+    #   Product.where(["price = %?", price]).sole
+    def sole
+      found, undesired = first(2)
+
+      case
```

Comment:
out of curiosity @kivikakk why write this as a `case` statement rather than `if found.nil?`, `elsif undesired.present?`?

Reply:
Good point. I already fixed it on master.

Reply:
Came from @dhh’s code sample. I pretty much always find case/when clearer than elsif since the conditions are aligned, same here.

Reply:
Ahhh, I missed that. The alignment of conditions is cool - I didn't notice that. I guess the trade off is that the if/elsif style reads like english, whereas (to me at least) a `case` with no subject after it looks broken.